io/src/vtk_lib_io.cpp

```diff
@@ -74,26 +74,35 @@ pcl::io::loadPolygonFile (const std::string &file_name, pcl::PolygonMesh& mesh)
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-int
-pcl::io::savePolygonFile (const std::string &file_name, const pcl::PolygonMesh& mesh)
+bool
+pcl::io::savePolygonFile (const std::string &file_name,
+                          const pcl::PolygonMesh& mesh,
+                          const bool binary_format)
 {
-  // TODO: what about binary/ASCII modes?!?!?!
   // TODO: what about sensor position and orientation?!?!?!?
-  // TODO: how to adequately catch exceptions thrown by the vtk writers?!
   std::string extension = file_name.substr (file_name.find_last_of (".") + 1);
   if (extension == "pcd") // no Polygon, but only a point cloud
   {
-    int error_code = pcl::io::savePCDFile (file_name, mesh.cloud);
+    int error_code;
+    if (binary_format)
+    {
+      pcl::PointCloud<pcl::PointXYZRGB> cloud;
```

Comment:
Hm, seems like my comment on this line disappeared somehow. Repeating: why this specific point type?